arangod/RestHandler/RestCursorHandler.cpp

```diff
@@ -775,7 +775,8 @@ void RestCursorHandler::lookupCursor(std::string_view id,
 
   TRI_ASSERT(_cursor != nullptr);
 
-  if (mustBeRetriable && !_cursor->isRetriable()) {
+  if (batchId.has_value() && _cursor->isCurrentBatchId(batchId.value()) &&
```

Comment:
Why only fail if the batchId is identical to the current batch id?
What if a batchId is specified that is totally out of range, but not identical to the current batch id? For example, if the batch id to fetch is 23 and you are specifying a batchId value of 1 or 99?